File: golden_ticket_enterprise/lib/entities/faq.dart
import 'package:golden_ticket_enterprise/entities/main_tag.dart';
import 'package:golden_ticket_enterprise/entities/sub_tag.dart';

class FAQ {
  final int faqID;
  final String title;
  final String description;
  final String solution;
  final DateTime createdAt;
  final bool isArchived;
  MainTag? mainTag;
  SubTag? subTag;
  FAQ({ required this.faqID, required this.title, required this.description, required this.solution, required this.createdAt, required this.isArchived, this.mainTag, this.subTag});

  factory FAQ.fromJson(Map<String, dynamic> json) {
    return FAQ(
        faqID: json['faqID'],
        title: json['title'],
        description: json['description'],
        solution:  json['solution'],
        createdAt: DateTime.parse(json['createdAt']),
        isArchived: json['isArchived'],
        mainTag: MainTag.fromJson(json['mainTag']),
        subTag: SubTag.fromJson(json['subTag'])
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'faqID': faqID,
      'title': title,
      'description': description,
      'solution':  solution,
      'createdAt': createdAt,
      'isArchived': isArchived,
      'mainTag': mainTag,
      'subTag': subTag
    };
  }

}

File: GoldenTicket/GoldenTicket/Entities/FAQ.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace GoldenTicket.Entities
{
    [Table("tblFAQ")]
    public class FAQ
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int FaqID { get; set; }
        [Required]
        public string? Title { get; set; }
        [Required]
        public string? Description { get; set; }
        [Required]
        public string? Solution { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public bool IsArchived { get; set; }
        [Required]
        public int? MainTagID { get; set;}
        [ForeignKey("MainTagID")]
        public MainTag? MainTag { get; set; }

        [Required]
        public int? SubTagID { get; set;}
        [ForeignKey("SubTagID")]
        public SubTag? SubTag { get; set; }

    }

    public class FAQDTO
    {
        public int FaqID { get; set; }
        public string? Title { get; set; }
        public string? Description { get; set; }
        public string? Solution { get; set; }
        public DateTime CreatedAt { get; set; }
        public bool IsArchived { get; set; }
        public MainTagDTO? MainTag { get; set; }
        public SubTagDTO? SubTag { get; set; }
    }

}
